nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -190,6 +190,17 @@ def __init__(self, cfg: DictConfig, trainer: Trainer, no_lm_init=True):
             gc.disable()
             self.validation_global_step = 1
 
+        self._val_micro_batches_consumed = 0
+
+    def _reconfigure_val_batches(self):
+        """
+        Reconfigure trainer.limit_val_batches for pretraining
+        """
+        # Override limit_val_batches to be a multiple of num microbatches and so there are limit_val_batches//num_micro_batches num of global batches
+        self.trainer.limit_val_batches *= get_num_microbatches()
+        # Override num sanity steps equal to num of microbatches and perform one val_step
+        self.trainer.num_sanity_val_steps = get_num_microbatches()
+
```

Comment:
This method is called only in pretraining models. Not setting `limit_val_batches` to be a multiple of `microbatches` for downstream models as we are passing just 1 batch to `val_step` each time and limit is not required. Val step is run as many times as the user entered `limit_val_batches` in case of downstream tasks. Also leaving the `num_sanity_val_step` for downstream at the default val of 2.